resolwe_bio/processes/WGBS/walt.yml

```diff
@@ -144,38 +166,58 @@
             -o "${NAME}_sorted.bam" \
             "${NAME}_mapped.bam"
       re-checkrc "Samtools sort command failed."
-      
-      samtools index "${NAME}_sorted.bam" "${NAME}_sorted.bam.bai"
-      re-checkrc "Samtools index command failed."
 
-      to-mr -o "${NAME}.mr" -m general "${NAME}_mapped.bam"
+      to-mr -o "${NAME}_mapped.mr" -m general "${NAME}_mapped.bam"
       re-checkrc "Conversion of bam to mr failed."
 
       LC_ALL=C \
       sort \
         -k 1,1 -k 2,2n -k 3,3n -k 6,6 \
         -S{{ requirements.resources.memory }}M \
         -o "${NAME}_sorted.mr" \
-        "${NAME}.mr"
+        "${NAME}_mapped.mr"
       re-progress 0.8
 
       {% if rm_dup %}
-        duplicate-remover -o "${NAME}.mr" "${NAME}_sorted.mr"
+        duplicate-remover -o "${NAME}_dremove.mr" "${NAME}_sorted.mr"
         re-checkrc "Removing duplicates failed."
       {% else %}
-        mv "${NAME}_sorted.mr" "${NAME}.mr"
+        mv "${NAME}_sorted.mr" "${NAME}_dremove.mr"
+      {% endif %}
+
+      {% if spikein_options.spikein_name is defined %}
+        grep ^{{ spikein_options.spikein_name }} "${NAME}_dremove.mr" > "${NAME}_spikein.mr"
+        re-checkrc "Given unmethylated control sequence name was not found."
+        pigz "${NAME}_spikein.mr"
+        re-save-file spikein_mr "${NAME}_spikein.mr.gz"
       {% endif %}
 
+      {% if spikein_options.spikein_name is defined and spikein_options.filter_spikein %}
+        samtools index "${NAME}_sorted.bam" "${NAME}_sorted.bam.bai"
+        samtools idxstats "${NAME}_sorted.bam" | \
+            cut -f 1 | grep -v {{ spikein_options.spikein_name }} | \
+            xargs samtools view -o "${NAME}.bam" "${NAME}_sorted.bam"
+        re-checkrc "Removal of unmethylated reads from BAM file failed."
+        grep -v ^{{ spikein_options.spikein_name }} "${NAME}_dremove.mr" > "${NAME}.mr"
+        re-checkrc "Removal of unmethylated reads from MR file failed."
```

Comment:
Should this rather be `re-checkrc "Removal of spike-in reads from MR file failed."`?